tsdb/engine/tsm1/compact.go

```diff
@@ -73,6 +73,13 @@ type DefaultPlanner struct {
 
 	// lastPlanCheck is the last time Plan was called
 	lastPlanCheck time.Time
+
+	mu sync.RWMutex
+	// lastFindGenerations is the last time findGenerations was run
+	lastFindGenerations time.Time
+
```

Comment:
nit: since `lastGenerations` is protected by `mu`, the blank line could be removed so it's in the same block as `lastFindGenerations`.